### PR TITLE
Follow-up fixes

### PR DESCRIPTION
**Pull request by kiryldz**
1. Use Double.compareTo and Float.compareTo when comparing these types in overloaded equals.
2. Builder setter methods should take into account whether given property is nullable. Now we always set nullable property and throw runtime exception when build() is called - this should be replaced with compile error.
3. If parameter is non-nullable and has no default value - it becomes Builder constructor parameter.
4. Javadoc looks nicer.

### CODE REVIEW

File: processor/src/main/kotlin/com/tobrun/datacompat/DataCompatProcessor.kt [hunk 1]
```diff
@@ -233,7 +230,24 @@ class DataCompatProcessor(
...
                             separator = "·&& ",
-                            transform = { "$it·==·other.$it" },
+                            transform = {
+                                val resolvedType = it.type.resolve()
+                                val isFloat =
+                                    resolvedType.toClassName() == Float::class.asTypeName()
+                                val isDouble =
+                                    resolvedType.toClassName() == Double::class.asTypeName()
+                                if (isFloat || isDouble) {
+                                    if (resolvedType.isMarkedNullable && isDouble) {
```
Comment (alexander-kulikovskii) [inline processor/src/main/kotlin/com/tobrun/datacompat/DataCompatProcessor.kt]: nit:
replace with `with(it) { ...}`

File: processor/src/main/kotlin/com/tobrun/datacompat/DataCompatProcessor.kt [hunk 2]
```diff
@@ -233,7 +230,24 @@ class DataCompatProcessor(
                         propertyMap.keys.joinToString(
                             prefix = "return ",
                             separator = "·&& ",
-                            transform = { "$it·==·other.$it" },
+                            transform = {
+                                val resolvedType = it.type.resolve()
+                                val isFloat =
+                                    resolvedType.toClassName() == Float::class.asTypeName()
+                                val isDouble =
+                                    resolvedType.toClassName() == Double::class.asTypeName()
```
Comment (alexander-kulikovskii) [inline processor/src/main/kotlin/com/tobrun/datacompat/DataCompatProcessor.kt]:
```suggestion
                                val isFloatOrDouble =
                                    resolvedType.toClassName() in setOf(Float::class.asTypeName(),
Double::class.asTypeName())
```

**Review by yunikkk — APPROVED** [review]

File: processor/src/main/kotlin/com/tobrun/datacompat/DataCompatProcessor.kt [hunk 3]
```diff
@@ -327,46 +363,36 @@ class DataCompatProcessor(
                         )
                         .addKdoc(
                             """
-                            |Set $kDocProperty
-                            |${property.key.docString?.trimStart(' ', '\n') ?: ""}
-                            |@param $propertyName $kDocProperty
+                            |Setter for ${property.value.kDoc.trimEnd('.')
```
Comment (jush) [inline processor/src/main/kotlin/com/tobrun/datacompat/DataCompatProcessor.kt]: Nit: This doesn't always work. For example:
```
* Setter for defines text color of the attribution icon.
```

Would it be too redundant to do:
```suggestion
                            |Setter for $propertyName: ${property.value.kDoc.trimEnd('.')
```

Comment (kiryldz) [reply processor/src/main/kotlin/com/tobrun/datacompat/DataCompatProcessor.kt]: Yeah, that slipped away from me 😄 will fix

**Review by jush — APPROVED** [review]

File: processor/src/main/kotlin/com/tobrun/datacompat/DataCompatProcessor.kt (hunk 2 above)
Comment (kiryldz) [reply processor/src/main/kotlin/com/tobrun/datacompat/DataCompatProcessor.kt]: I have separate checks for `isMarkedNullable && isDouble` and `isMarkedNullable && isFloat` so I need separate variables.

File: processor/src/main/kotlin/com/tobrun/datacompat/DataCompatProcessor.kt (hunk 1 above)
Comment (kiryldz) [reply processor/src/main/kotlin/com/tobrun/datacompat/DataCompatProcessor.kt]: done (if I understood the suggestion correctly)

File: processor/src/main/kotlin/com/tobrun/datacompat/DataCompatProcessor.kt [hunk 4]
```diff
@@ -233,7 +230,25 @@ class DataCompatProcessor(
...
+                                            "($it·?:·0.0).compareTo(other.$it·?:·0.0)·==·0"
+                                        } else if (isMarkedNullable && isFloat) {
+                                            "($it·?:·0f).compareTo(other.$it·?:·0f)·==·0"
+                                        } else {
+                                            "$it.compareTo(other.$it)·==·0"
+                                        }
+                                    } else {
+                                        "$it·==·other.$it"
+                                    }
+                                }
```
Comment (alexander-kulikovskii) [inline processor/src/main/kotlin/com/tobrun/datacompat/DataCompatProcessor.kt]:
```suggestion
                               val isFloat =
                                        toClassName() == Float::class.asTypeName()
                                    val isDouble =
                                        toClassName() == Double::class.asTypeName()
                                when(it.type.resolve()) {
                                    is (isDouble.not() && isFloat.not()) -> "$it.compareTo(other.$it)·==·0"
                                    is (isMarkedNullable && isDouble) -> "($it·?:·0.0).compareTo(other.$it·?:·0.0)·==·0"
                                     is (isMarkedNullable && isFloat) -> "($it·?:·0f).compareTo(other.$it·?:·0f)·==·0"
                                     else ->"$it·==·other.$it"
                                }
```

**Review by alexander-kulikovskii — APPROVED** [review]

File: processor/src/main/kotlin/com/tobrun/datacompat/DataCompatProcessor.kt (hunk 4 above)
Comment (kiryldz) [reply processor/src/main/kotlin/com/tobrun/datacompat/DataCompatProcessor.kt]: done with some changes

**Pull request merged**